worker/worker.go

```diff
@@ -702,6 +703,64 @@ func (w *worker) rhpRegistryUpdateHandler(jc jape.Context) {
 	}
 }
 
+func (w *worker) rhpSyncHandler(jc jape.Context) {
+	ctx := jc.Request.Context()
+	var rsr api.RHPSyncRequest
+	if jc.Decode(&rsr) != nil {
+		return
+	}
+
+	// Get IP of host.
+	h, err := w.bus.Host(ctx, rsr.HostKey)
+	if jc.Check("failed to fetch host", err) != nil {
+		return
+	}
+	hostIP := h.Settings.NetAddress
+	siamuxAddr := h.Settings.SiamuxAddr()
+
+	// Get contract revision.
+	lockID, err := w.bus.AcquireContract(jc.Request.Context(), rsr.ContractID, lockingPriorityFunding, lockingDurationFunding)
```

Comment:
I think it's kind of weird the `RHPSyncRequest` contains the contract ID, especially since we go and fetch the host IP. Can't we simply fetch the active contract for the host as well? Endpoint feels weird having to specify the contract ID, seems like an extra step if you have to do it manually.